docker/docker-upgrade

```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 set -e
-# based on https://github.com/tianon/docker-postgres-upgrade/blob/master/12-to-14/docker-upgrade
+# based on https://github.com/tianon/docker-postgres-upgrade/blob/master/12-to-16/docker-upgrade
```

Comment:
in the source repo 12 to 14 is exactly the same as 12 to 16. But since this 12 to 14 wasn't exactly the same as in the source repo (the additional check if owner is postgres) I've kept it as is